panel/widgets/button.py

```diff
@@ -125,28 +129,62 @@ class Button(_ClickButton):
     def _linkable_params(self):
         return super()._linkable_params + ['value']
 
-    def jslink(self, target, code=None, args=None, bidirectional=False, **links):
+    def jslink(self, target: 'JSLinkTarget', code: Dict[str, str]=None, args: Optional[Dict]=None, bidirectional: bool=False, **links: str) -> 'Link':
+        """
+        Links properties on the this Button to those on the
+        `target` object in Javascript (JS) code.
+        
+        Supports two modes, either specify a
+        mapping between the source and target model properties as
+        keywords or provide a dictionary of JS code snippets which
+        maps from the source parameter to a JS code snippet which is
+        executed when the property changes.
+
+        Arguments
+        ----------
+        target: panel.viewable.Viewable | bokeh.model.Model | holoviews.core.dimension.Dimensioned 
+          The target to link the value(s) to.
+        code: dict
+          Custom code which will be executed when the widget value
+          changes.
+        args: dict
+          A mapping of objects to make available to the JS callback
+        bidirectional: boolean
+          Whether to link source and target bi-directionally. Default is `False`.
+        **links: dict[str,str]
+          A mapping between properties on the source model and the
+          target model property to link it to.
+
+        Returns
+        -------
+        Link
+          The Link can be used unlink the widget and the target model.
+        """
         links = {'event:'+self._event if p == 'value' else p: v for p, v in links.items()}
         super().jslink(target, code, args, bidirectional, **links)
 
-    jslink.__doc__ = Widget.jslink.__doc__
-
     def _process_event(self, event):
         self.param.trigger('value')
         self.clicks += 1
 
-    def on_click(self, callback):
+    def on_click(self,
+    callback: Callable[[param.parameterized.Event], None]) -> param.parameterized.Watcher:
         """
         Register a callback to be executed when the `Button` is clicked.
 
         The callback is given an `Event` argument declaring the number of clicks
 
         Arguments
         ---------
-        callback: (callable)
+        callback: (Callable[[param.parameterized.Event], None])
             The function to run on click events. Must accept a positional `Event` argument
+
+        Returns
+        -------
+        watcher: param.Parameterized.Watcher
+          A `Watcher` that executes the callback when the button is clicked.
         """
-        self.param.watch(callback, 'clicks', onlychanged=False)
+        return self.param.watch(callback, 'clicks', onlychanged=False)
```

Comment:
I updated the function to return the `Watcher` I believe this will not break anything and its more useful as you then have a chance to later delete/ stop/ pause the `Watcher`.

Reply:
Sounds good!